skin/src/main/java/org/researchstack/skin/task/SmartSurveyTask.java

```diff
@@ -116,8 +122,13 @@ else if(type.equals("DateConstraints"))
         {
             answerFormat = new DateAnswerFormat(AnswerFormat.DateAnswerStyle.Date);
         }
+        else if(type.equals("DurationConstraints"))
+        {
+            answerFormat = new DurationAnswerFormat(constraints.step, constraints.durationUnit);
+        }
         else
         {
+            LogExt.e(SmartSurveyTask.class, "Survey question has answer type not supported:" + type);
```

Comment:
This is useful -- thanks for adding it! But how do you feel about going the "stronger" route and actually raising an exception here, or at least a warning? I feel like it's better to catch unsupported question types here with an explicit exception than let this log message get buried in the stack trace somewhere. Thoughts? @dariosalviwork @nrlakin @crayne